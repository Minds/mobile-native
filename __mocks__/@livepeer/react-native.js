// mock @livepeer/react-native
export const LivepeerConfig = {
  setConfig: jest.fn(),
  getConfig: jest.fn(),
  get: jest.fn(),
  set: jest.fn(),
};

export const createReactClient = jest.fn();
export const createLivepeerClient = jest.fn();
export const Player = jest.fn();
