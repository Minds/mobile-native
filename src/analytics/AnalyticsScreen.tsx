import { observer } from 'mobx-react';
import React, { useCallback, useState } from 'react';
import { Text, View } from 'react-native';
import TopbarTabbar from '../common/components/topbar-tabbar/TopbarTabbar';
import i18n from '../common/services/i18n.service';
import ThemedStyles from '../styles/ThemedStyles';
import DashboardTab from './tabs/dashboard/DashboardTab';
import TokensTab from './tabs/tokens/TokensTab';
import TrendingTab from './tabs/trending/TrendingTab';

type TAnalyticsTabs =
  | 'earnings'
  | 'engagement'
  | 'traffic'
  | 'trending'
  | 'tokens';

interface AnalyticsScreenProps {
  navigation: any;
}

const AnalyticsScreen = observer(({ navigation }: AnalyticsScreenProps) => {
  const theme = ThemedStyles.style;
  const [activeTabId, setActiveTabId] = useState<TAnalyticsTabs>('engagement');
  const _onTabBarChange = useCallback(
    (id: string) => setActiveTabId(id as TAnalyticsTabs),
    [],
  );

  const screen = () => {
    switch (activeTabId) {
      case 'earnings':
        return (
          <DashboardTab
            key={'earnings'}
            url={'api/v2/analytics/dashboards/earnings'}
            defaultMetric={'earnings_total'}
          />
        );
      case 'engagement':
        return (
          <DashboardTab
            key={'engagement'}
            url={'api/v2/analytics/dashboards/engagement'}
            defaultMetric={'votes_up'}
          />
        );
      case 'traffic':
        return (
          <DashboardTab
            key={'traffic'}
            url={'api/v2/analytics/dashboards/traffic'}
            defaultMetric={'page_views'}
          />
        );
      case 'trending':
        return <TrendingTab navigation={navigation} />;
      case 'tokens':
        return <TokensTab />;
      default:
        return <View />;
    }
  };

  return (
    <View style={theme.flexContainer}>
      <Text style={[theme.padding4x, theme.titleText, theme.paddingBottom2x]}>
        {i18n.t('analytics.title')}
      </Text>
      <TopbarTabbar
        current={activeTabId}
        onChange={_onTabBarChange}
        tabs={[
          { id: 'traffic', title: i18n.t('analytics.traffic') },
          { id: 'tokens', title: i18n.t('analytics.tokens.title') },
          { id: 'engagement', title: i18n.t('analytics.engagement') },
          { id: 'earnings', title: i18n.t('analytics.earnings') },
          { id: 'trending', title: i18n.t('analytics.trending.title') },
        ]}
      />
      <View style={theme.centered}>{screen()}</View>
    </View>
  );
});

export default AnalyticsScreen;
