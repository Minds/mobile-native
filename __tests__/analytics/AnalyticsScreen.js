import { shallow } from 'enzyme';
import React from 'react';
import 'react-native';

// Note: test renderer must be required after react-native.
import AnalyticsScreen from '../../src/analytics/AnalyticsScreen';

/**
 * Tests
 */
describe('AnalyticsScreen component', () => {
  it('should render correctly', () => {
    const component = shallow(<AnalyticsScreen />);

    expect(component).toMatchSnapshot();
  });
});
