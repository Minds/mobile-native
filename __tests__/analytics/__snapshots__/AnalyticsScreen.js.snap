// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnalyticsScreen component should render correctly 1`] = `
<View>
  <Text
    style={
      Array [
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    Analytics
  </Text>
  <TopbarTabbar
    current="engagement"
    onChange={[Function]}
    tabs={
      Array [
        Object {
          "id": "traffic",
          "title": "Traffic",
        },
        Object {
          "id": "tokens",
          "title": "Tokens",
        },
        Object {
          "id": "engagement",
          "title": "Engagement",
        },
        Object {
          "id": "earnings",
          "title": "Earnings",
        },
        Object {
          "id": "trending",
          "title": "Trending",
        },
      ]
    }
  />
  <View>
    <Memo(wrappedComponent)
      defaultMetric="votes_up"
      key="engagement"
      url="api/v2/analytics/dashboards/engagement"
    />
  </View>
</View>
`;
